src/theme/palette.ts

```diff
@@ -6,24 +6,163 @@
 import { alpha, PaletteOptions } from '@mui/material';
 
 // Palette and PalleteOptions types can be extended here as needed
-/*
+
 declare module '@mui/material/styles/createPalette' {
   interface Palette {
+    buy: Palette['primary'];
```

Comment:
Did you try using `PaletteColorOptions` rather than `Palette['primary']`.  I think the type is named and exported.

Using `Palette['primary']` sets the buy type to the type of `Palette['primary']` which should be `PaletteColorOptions`, so this is doing the same thing, but better to directly use the type if possible.

Reply:
Yea, make sense. I found a type export, [https://github.com/mui/material-ui/blob/89687f38cae750650555772ba4d821c9084d8dfc/packages/material-ui/src/styles/createPalette.d.ts](url), so based on this, inside interface `Palette` should be` buy: PaletteColor` and inside `PaletteOptions` should be` buy: PaletteColorOptions`, right?